### PR TITLE
Update Proxy SHA to latest with TCP proxy fixes.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Pulling the following changes from github.com/istio/proxy:

f498337 Fix a bug in origin authenticator that wrongly treats empty origin methods as pass (#1962)
c352de0 Mixer Client: Add support for TCP local attributes (#1967)
2c563c6 remove not used path patcher functions (#1966)
490d26f Update Envoy SHA to latest with TCP proxy fixes. (#1964)
4cc4b7c Mixer Client uses Node metadata to populate Mixer attributes (#1961)
cf23357 Fix the peerIsOptional and originIsOptional for authn filter. (#1959)
cc6e58e support per-path JWT validation. (#1928)
a5dd1aa mixer: clear route cache on header update (#1946)

Pulling the following changes from github.com/envoyproxy/envoy:

f936fc60f ssl: serialize accesses to SSL socket factory contexts (#4345)
e34dcd62a Fix crash in tcp_proxy (#4323)
ae6a25222 router: fix matching when all domains have wildcards (#4326)
aa06142ff test: Stop fake_upstream methods from accidentally succeeding (#4232)
5d731878f rbac: update the authenticated.user to a StringMatcher. (#4250)
c6bfc7d9a time: Event::TimeSystem abstraction to make it feasible to inject time with simulated timers (#4257)
752483ea9 Fixing the fix (#4333)
83487f6f3 tls: update BoringSSL to ab36a84b (3497). (#4338)
7bc210e02 test: fixing interactions between waitFor and ignore_spurious_events (#4309)
69474b398 admin: order stats in clusters json admin (#4306)
2d155f901 ppc64le build (#4183)
07efc6dc6 fix static initialization fiasco problem (#4314)
0b7e3b5e0 test: Remove declared but undefined class methods (#4297)
1485a1304 lua: make sure resetting dynamic metadata wrapper when request info is marked dead
d243cd62e test: set to zero when start_time exceeds limit (#4328)
0a1e92acc test: fix heap use-after-free in ~IntegrationTestServer. (#4319)
cddc732c7 CONTRIBUTING: Document 'kick-ci' trick. (#4335)
f13ef2464 docs: remove reference to deprecated value field (#4322)
e947a2766 router: minor doc fixes in stream idle timeout (#4329)
0c2e998af tcp-proxy: fixing a TCP proxy bug where we attempted to readDisable a closed connection (#4296)
00ffe44a2 utility: fix strftime overflow handling. (#4321)
af1183c28 Re-enable TcpProxySslIntegrationTest and make the tests pass again. (#4318)
35534617b fuzz: fix H2 codec fuzzer post #4262. (#4311)
42f604853 Proto string issue fix (#4320)
9c492a01d Support Envoy to fetch secrets using SDS service. (#4256)
a8572192f ratelimit: revert `revert rate limit failure mode config` and add tests (#4303)
1d34172bd dns: fix exception unsafe behavior in c-ares callbacks. (#4307)
121242340 alts: add gRPC TSI socket (#4153)
f0363ae63 fuzz: detect client-side resets in H2 codec fuzzer. (#4300)
01aa3f820 test: hopefully deflaking echo integration test (#4304)
1fc0f4ba2 ratelimit: link legacy proto when message is being used (#4308)
aa4481e6b fix rare List::remove(&target) segfault (#4244)
89e0f23ba headers: fixing fast fail of size-validation (#4269)
97eba5918 build: bump googletest version. (#4293)
0057e22d9 fuzz: avoid false positives in HCM fuzzer. (#4262)
9d094e590 Revert ac0bd74f6f9716e3a44d1412f795317c30ca770a (#4295)
ddb28a4a1 Add validation context provider (#4264)
3b47cbabb added histogram latency information to Hystrix dashboard stream (#3986)
cf87d50cd docs: update SNI FAQ. (#4285)
f952033a4 config: fix update empty stat for eds (#4276)
329e591d3 router: Add ability of custom headers to rely on per-request data (#4219)
68d20b46c  thrift: refactor build files and imports (#4271)
5fa8192a3 access_log: log requested_server_name in tcp proxy (#4144)
fa45bb48f fuzz: libc++ clocks don't like nanos. (#4282)
53f8944f7 stats: add symbol table for future stat name encoding (#3927)
c987b425b test infra: Remove timeSource() from the ClusterManager api (#4247)
cd171d9a9 websocket: tunneling websockets (and upgrades in general) over H2 (#4188)
b9dc5d9a0 router: disallow :path/host rewriting in request_headers_to_add. (#4220)
0c9101127 network: skip socket options and source address for UDS client connections (#4252)
da1857d59 build: fixing a downstream compile error by noting explicit fallthrough (#4265)
9857cfe2a fuzz: cleanup per-test environment after each fuzz case. (#4253)
52beb067d test: Wrap proto string in std::string before comparison (#4238)
f5e219edc extensions/thrift_proxy: Add header matching to thrift router (#4239)
c9ce5d2b1 fuzz: track read_disable_count bidirectionally in codec_impl_fuzz_test. (#4260)
35103b353 fuzz: use nanoseconds for SystemTime in RequestInfo. (#4255)
ba6ba9883 fuzz: make runtime root hermetic in server_fuzz_test. (#4258)
b0a901480 time: Add 'format' test to ensure no one directly instantiates Prod*Time from source. (#4248)
85674603b access_log: support beginning of epoch in START_TIME. (#4254)
28d5f4118 proto: unify envoy_proto_library/api_proto_library. (#4233)
f7d3cb638 http: fix allocation bug introduced in #4211. (#4245)

Fixes istio/istio#8310.

Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

File: mixer/test/client/check_cache_hit/check_cache_hit_test.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 const checkAttributesOkGet = `
 {
   "context.protocol": "http",
+  "context.reporter.uid": "",
```
Comment (PiotrSikora) [inline mixer/test/client/check_cache_hit/check_cache_hit_test.go]: @mandarjog should all those be empty?

**Comment by yangminzhu** [comment]
@PiotrSikora 
I have a fix for that in https://github.com/istio/proxy/pull/1962. Would you mind to wait a little bit and update your proxy sha to include that fix? (I'm also about to update the proxy sha)

**Comment by PiotrSikora** [comment]
@yangminzhu Sounds good, I'll bump the SHA once your changes are merged!

**Comment by yangminzhu** [comment]
@PiotrSikora 
I see you just updated the Envoy sha in Istio proxy to https://github.com/envoyproxy/envoy/pull/4345. It then includes a breaking change (https://github.com/envoyproxy/envoy/pull/4250) for the RBAC API in Envoy.

This means we need to do some extra work when updating the proxy sha in this PR

* Update the github.com/envoyproxy/go-control-plane revision to past `a29483497d9342d4e6419dd99723cb40fae96670`, I already have a PR: https://github.com/envoyproxy/go-control-plane/pull/113 waiting for review.

* Update the istio/istio repo to use the updated go-control-plane commit, this should be simple (hoping there is no other breaking changes)

* Update the [rbac.go](https://github.com/istio/istio/blob/master/pilot/pkg/networking/plugin/authz/rbac.go). The required code change is small, should be < 10 lines. I can create a PR with required changes so that you can simply copy to this PR. I'll do this once my above PR is merged.

Sorry for the inconveniences.

**Comment by yangminzhu** [comment]
@PiotrSikora 
I have created a sample PR in https://github.com/istio/istio/pull/8698. There are two commits in the PR, the first commit is doing the same thing as you did here, the [second commit](https://github.com/istio/istio/pull/8698/commits/d009ee804a7598aaf029f3282050ea7c63f730db) is fixing some additional build/test fails that you should copy to this PR as well.

**Comment by JimmyCYJ** [comment]
Hi @PiotrSikora , is it convenient for you to update proxy SHA to 9dd2de3bcf0e37febfc70680db8142916b8c6099
(https://github.com/istio/proxy/pull/1975)?

**Comment by PiotrSikora** [comment]
@JimmyCYJ No, it takes about a week to update the proxy, and there are usually more changes required than simply updating the SHA, so let's merge this PR as-is instead of delaying it even more.

Once it's merged, feel free to open another PR.

**Pull request merged**